x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/feature/SimpleFeatureFactory.java

```diff
@@ -77,7 +77,8 @@ public SimpleFeatureFactory(int z, int x, int y, int extent) {
             if (posLat > extent || posLat < 0) {
                 continue;
             }
-            if (i == 0 || posLon != prevLon || posLat != prevLat) {
+            // filter out repeated points
+            if (numPoints == 0 || posLon != prevLon || posLat != prevLat) {
```

Comment:
Since you are no longer using `i`, you can probably replace the for loop and remove it completely.